car-pool/car-mirror/http.md

```diff
@@ -77,25 +77,25 @@ This field MUST NOT be interpreted as a CID root being sent.
 
 ## 2.2 Requestor Payload
 
-```cddl
-payload = {
-  bk : uint,  ; Bloom filter hash count
-  bb : bytes, ; Bloom filter Binary
-  pl : car,   ; Data payload
-}
+```ipldsch
+type ReqPayload struct {
+  bk Integer -- Bloom filter hash count
+  bb Bytes   -- Bloom filter Binary
+  pl CarFile -- Data payload
+} representation tuple
```

Comment:
This is probably the most controversial part of this spec. IPLD Schema has [representation strategies](https://ipld.io/docs/schemas/features/representation-strategies/#struct-tuple-representation) for strcuts (and a couple other types).

With the `map` representation, IPLD will default canonicalize the order of fields. We want to impose order, hence the `tuple` (array) representation. As a bonus, it's now ordered, so we don't strictly speaking need those keys. Better yet: if you deserialize this with an IPLD codec (e.g. `dag-cbor`), you'll parse out a struct with the fields on it.

Reply:
Do we need the keys to be 2-character then?
I guess at least for consistency with `PullRequest` for `bk` and `bb`. But perhaps `payload` instead of `pl`?
Also, should `PullRequest` be `representation tuple`, too?

Reply:
Yeah this is what I meant by controversial: this being the only tulle. What I *actually* want is a `representation orderedmap` or similar IPLD schema, especially for extensibility. In fact, maybe we just add that to the prose -- CBOR itself supports it, this just changes the CID of the message, which we don't care about here.